pkg/analytics/up.go

```diff
@@ -0,0 +1,120 @@
+package analytics
+
+import (
+	"time"
+
+	"github.com/okteto/okteto/pkg/model"
+)
+
+const (
+	// Event that tracks when a user activates a development container
+	upEvent = "Up"
+)
+
+// UpMetadata defines the properties an up can have
+type UpMetadata struct {
+	isV2                   bool
+	manifestType           model.Archetype
+	isInteractive          bool
+	isOktetoRepository     bool
+	hasDependenciesSection bool
+	hasBuildSection        bool
+	hasDeploySection       bool
+	hasReverse             bool
+	isHybridDev            bool
+	mode                   string
+	failActivate           bool
+	activateDuration       time.Duration
+	initialSyncDuration    time.Duration
+	isReconnect            bool
+	reconnectCause         string
+	errSync                bool
+	errResetDatabase       bool
+	success                bool
+}
+
+func NewUpMetadata() *UpMetadata {
+	return &UpMetadata{}
+}
+
+func (u *UpMetadata) toProps() map[string]interface{} {
+	return map[string]interface{}{
+		"isInteractive":              u.isInteractive,
+		"isV2":                       u.isV2,
+		"manifestType":               u.manifestType,
+		"isOktetoRepository":         u.isOktetoRepository,
+		"hasDependenciesSection":     u.hasDependenciesSection,
+		"hasBuildSection":            u.hasBuildSection,
+		"hasDeploySection":           u.hasDeploySection,
+		"hasReverse":                 u.hasReverse,
+		"mode":                       u.mode,
+		"failActivate":               u.failActivate,
+		"activateDurationSeconds":    u.activateDuration.Seconds(),
+		"initialSyncDurationSeconds": u.initialSyncDuration.Seconds(),
+		"isReconnect":                u.isReconnect,
+		"reconnectCause":             u.reconnectCause,
+		"errSync":                    u.errSync,
+		"errResetDatabase":           u.errResetDatabase,
+	}
+}
+
+func (u *UpMetadata) AddManifestProps(m *model.Manifest) {
+	u.isV2 = m.IsV2
+	u.manifestType = m.Type
+	u.hasDependenciesSection = m.HasDependenciesSection()
+	u.hasBuildSection = m.HasBuildSection()
+	u.hasDeploySection = m.HasDeploySection()
+}
+
+func (u *UpMetadata) AddDevProps(d *model.Dev) {
```

Comment:
Can we agree on the convention for all these names? I think it's odd to have add/Set/CommandSuccess I think all of them should follow the same structure since it's modifying the struct `UpMetadata` with the data from the function. 

Also if we add props to one I'd do it for every function, but I think it's better to remove the suffix